synapse/api/auth.py

```diff
@@ -30,7 +30,12 @@
 from synapse.appservice import ApplicationService
```

Comment:
> LGTM — thanks for doing this; just the other day I was talking about this kind of 'unstructured list of database calls' that we have in Jaeger and about how we would do well to structure some of them. Seems like you've made a decent dent into that for `/messages` at least.
>
> *-- @reivilibre, https://github.com/matrix-org/synapse/pull/13368#pullrequestreview-1059988209*

Thanks for the review @reivilibre!

I'm curious why we don't just auto-instrument every function? Is there any performance penalty? Bad practice? Some of it is noise but seems a lot useful to know the function calls than not have them.